tests/numpy/test_creation_manipulation.py

```diff
@@ -78,6 +79,55 @@ def test_diag(nps_app_inst):
     np_arr = np.diag(np_arr)
     assert np.allclose(ba.get(), np_arr)
 
+def test_trace(nps_app_inst):
+    import nums.numpy as nps
+
+    assert nps_app_inst is not None
+
+    a: BlockArray = nps.array([1.0, 2.0, 3.0, 4.0])
+
+    # Test that trace doesn't work with 1-dimensional vectors
+    with pytest.raises(ValueError):
+        nps.trace(a)
+
+    # Create Diagonal Matrices
+    a_diag = nps.diag(a) # Nums Version
+    a_diag_np = np.diag(a.get()) # Numpy version
+
+    # Apply on Trace on each of the Diagonal matrices
+    a_diag_trace = nps.trace(a_diag).get()
+    a_diag_np_trace = np.trace(a_diag_np)
+
+    assert np.allclose(a_diag_trace, a_diag_np_trace)
+
+    # Use pre-created diagonal matrices
+    b: BlockArray = nps.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]])
+
+    # Call trace
+    b_diag_trace = nps.trace(b).get() # Nums Version
+    b_diag_np_trace = np.trace(b.get()) # Numpy version
+
+    assert np.allclose(b_diag_trace, b_diag_np_trace)
+
+    # Test that trace doesn't work for 3+ axes
+    mat: BlockArray = nps.zeros((2, 3, 2))
+    with pytest.raises(ValueError):
+        nps.trace(mat)
+
+    # Test that trace doesn't work for non-zero offset
+    mat:  BlockArray = nps.array([1.0, 2.0, 3.0, 4.0])
+    mat_diag = nps.diag(mat)
+    with pytest.raises(NotImplementedError):
+        nps.trace(mat_diag, offset=2)
+
+    # Test data type of the return type
+    mat_diag = nps.diag(nps.array([1.01, 2.02, 3.03, 4.04]))
+    mat_diag_np = np.diag(np.array([1.01, 2.02, 3.03, 4.04]))
+    mat_diag_trace = nps.trace(mat_diag, dtype=int).get()
+    mat_diag_np_trace = np.trace(mat_diag_np, dtype=int)
+
+    assert np.allclose(mat_diag_trace, mat_diag_np_trace)
```

Comment:
This is good. Can we include a test that ensures that, if the input dtype is integer, the output dtype is also integer?

Reply:
Sure. Let me try that. Probably check the datatype of the first element or just find a way to get the type of the whole array.